.github/workflows/tests.yml

```diff
@@ -0,0 +1,115 @@
+          access_token: ${{ github.token }}
+      - uses: actions/checkout@v1
+      - uses: actions/setup-python@v1
+      - name: set PY
+        run: echo "::set-env name=PY::$(python -c 'import hashlib, sys;print(hashlib.sha256(sys.version.encode()+sys.executable.encode()).hexdigest())')"
+      - name: install clang-format
+        if: steps.clang_format.outputs.cache-hit != 'true'
+        run: |
+          sudo apt-get remove -y clang-6.0 libclang-common-6.0-dev libclang1-6.0 libllvm6.0
+          sudo apt-get autoremove
+          sudo apt-get install clang-format clang-format-6.0
+      - uses: pre-commit/action@v2.0.0
+
+  test:
+    name: Python
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python: [ 3.6, 3.7, 3.8 ]
+        os:  [ macos-latest, ubuntu-latest, windows-latest ]
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Cache conda and dependancies
+        id: cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            c:\Miniconda\envs\anaconda-client-env
+            /usr/local/miniconda/envs/anaconda-client-env
+            /usr/share/miniconda/envs/anaconda-client-env
+            ~/.profile
+          key: ${{ runner.os }}-${{ matrix.python}}-conda-v5-${{ hashFiles('python/requirements/CI-tests-conda/requirements.txt') }}-${{ hashFiles('python/requirements/CI-tests-pip/requirements.txt') }}
+
+      - name: Install Conda
+        uses: conda-incubator/setup-miniconda@v1
+        if: steps.cache.outputs.cache-hit != 'true'
+        with:
+          activate-environment: anaconda-client-env
+          python-version: ${{ matrix.python }}
+          channels: conda-forge
+          channel-priority: strict
+          auto-update-conda: true
+          use-only-tar-bz2: true # IMPORTANT: This needs to be set for caching to work properly!
+
+      - name: Fix windows symlinks
+        shell: bash
+        working-directory: python
+        if: matrix.os == 'windows-latest'
+        run: |
+          rm -r lib
+          mkdir lib
+          cp -r --dereference ../c/subprojects lib/.
+          cp -r --dereference ../c/tskit lib/.
+          cp ../c/tskit.h lib/.
+
+      - name: Fix windows .profile
+        if: steps.cache.outputs.cache-hit != 'true' && matrix.os == 'windows-latest'
+        shell: bash
+        run: |
+          cp ~/.bash_profile ~/.profile
+
+      - name: Install conda deps
+        if: steps.cache.outputs.cache-hit != 'true'
+        shell: bash -l {0}
+        run: conda install --yes --file=python/requirements/CI-tests-conda/requirements.txt
+
+      - name: Install pip deps
+        if: steps.cache.outputs.cache-hit != 'true'
+        shell: bash -l {0}
+        env:
+          HTSLIB_CONFIGURE_OPTIONS: --disable-bz2
```

Comment:
It's not obvious to me why we're messing with htslib here - would it be better to try and get pysam from bioconda? (Presumably I tried this before and failed). It's a bit obscure anyway, a comment might help here.

Reply:
You're right, wasn't needed.